dataplane/redbeams/redbeams.go

```diff
@@ -53,14 +52,120 @@ func ListRBDMSInstances(c *cli.Context) {
+	if err != nil {
+		commonutils.LogErrorAndExit(err)
+	}
+
+	server := resp.Payload
+	output := commonutils.Output{Format: c.String(fl.FlOutputOptional.Name)}
+	output.Write(listHeader, &redbeamsDetails{
+		Name:           *server.Name,
+		CRN:            server.Crn,
+		EnvironmentCrn: *server.EnvironmentCrn,
+	})
+}
+
+func GetDatabaseServerByName(c *cli.Context) {
+	defer commonutils.TimeTrack(time.Now(), "Get a database server by name")
+	envCrn := c.String(fl.FlEnvironmentCrn.Name)
+	name := c.String(fl.FlName.Name)
+	redbeamsDbServerClient := ClientRedbeams(*oauth.NewRedbeamsClientFromContext(c)).Redbeams.V4databaseservers
+
+	log.Infof("[GetDBServerByName] Getting database server with name: %s", name)
+	resp, err := redbeamsDbServerClient.GetDatabaseServerByName(v4databaseservers.NewGetDatabaseServerByNameParams().WithEnvironmentCrn(envCrn).WithName(name))
+	if err != nil {
+		commonutils.LogErrorAndExit(err)
+	}
 
+	server := resp.Payload
+	output := commonutils.Output{Format: c.String(fl.FlOutputOptional.Name)}
+	output.Write(listHeader, &redbeamsDetails{
+		Name:           *server.Name,
+		CRN:            server.Crn,
+		EnvironmentCrn: *server.EnvironmentCrn,
+	})
+}
+
+func GetDatabaseServerStatus(c *cli.Context) {
+	defer commonutils.TimeTrack(time.Now(), "Get the status of a database server by CRN")
+	crn := c.String(fl.FlCrn.Name)
 	redbeamsDbServerClient := ClientRedbeams(*oauth.NewRedbeamsClientFromContext(c)).Redbeams.V4databaseservers
+
+	log.Infof("[GetDBServerStatus] Getting status for database server with CRN: %s", crn)
+	resp, err := redbeamsDbServerClient.GetDatabaseServerStatusByCrn(v4databaseservers.NewGetDatabaseServerStatusByCrnParams().WithCrn(crn))
+	if err != nil {
+		commonutils.LogErrorAndExit(err)
+	}
+
+	status := resp.Payload
+	output := commonutils.Output{Format: c.String(fl.FlOutputOptional.Name)}
+	output.Write(listHeader, &redbeamsDetails{
+		Name:           *status.Name,
+		CRN:            *status.ResourceCrn,
+		EnvironmentCrn: *status.EnvironmentCrn,
+		Status:         *status.Status,
+	})
+}
+
+func GetDatabaseServerStatusByName(c *cli.Context) {
+	defer commonutils.TimeTrack(time.Now(), "Get the status of a database server by name")
+	envCrn := c.String(fl.FlEnvironmentCrn.Name)
+	name := c.String(fl.FlName.Name)
+	redbeamsDbServerClient := ClientRedbeams(*oauth.NewRedbeamsClientFromContext(c)).Redbeams.V4databaseservers
+
+	log.Infof("[GetDBServerStatusByName] Getting status for database server with name: %s", name)
+	resp, err := redbeamsDbServerClient.GetDatabaseServerStatusByName(v4databaseservers.NewGetDatabaseServerStatusByNameParams().WithEnvironmentCrn(envCrn).WithName(name))
+	if err != nil {
+		commonutils.LogErrorAndExit(err)
+	}
+
+	status := resp.Payload
+	output := commonutils.Output{Format: c.String(fl.FlOutputOptional.Name)}
+	output.Write(listHeader, &redbeamsDetails{
```

Comment:
Also here I would add all the other specs a databaseServerConfig has.
I might make a common method for listings in get and getByName, but might not be worth it, depends on the length.

Reply:
That all makes sense. I'll add some more to the details object.